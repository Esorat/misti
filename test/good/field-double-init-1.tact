contract Test {
    a: Int = 0;
    init(x: Int) {
        self.a = x; // Should be highlighted
    }
}
