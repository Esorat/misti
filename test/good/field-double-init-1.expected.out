test/good/field-double-init-1.tact:4:9:
  3 |     init(x: Int) {
> 4 |         self.a = x; // Should be highlighted
              ^
  5 |     }
Field Initialized Twice
Help: Consider initializing the field only in its declaration or in the `init` function
See: https://nowarp.github.io/tools/misti/docs/detectors/FieldDoubleInit